### PR TITLE
fix batch_norm amp autocast

### PR DESCRIPTION
**Pull request by yaochengji**
Fix https://github.com/pytorch/xla/issues/8496,

The test is put in a new class because the implementation of batch_norm amp is different from others (XLA requires the input dtypes of batch_norm to be all the same).

### CODE REVIEW

**Review by qihqi — APPROVED** [review]

**Comment by yaochengji** [comment]
@qihqi `tpu-ci` label was not added in this PR and the current CI keeps failing because of this.

Should we revert this PR and fixed the `tpu-ci` issue later?
